sparsify/trainer.py

```diff
@@ -702,6 +712,50 @@ def save_best(self, avg_loss: float | dict[str, float]):
         if dist.is_initialized():
             dist.barrier()
 
+    @torch.no_grad()
+    def evaluate(self, eval_dataset, step: int):
+        """Run eval set through spliced vs unspliced model,
+        compute mean KL divergence."""
+        if eval_dataset is None:
+            return
```

Comment:
Since we are doing a None check for the eval_dataset at the callsite on line 529 we can remove the None check here, and type evaluate's eval_dataset parameter as Dataset